test/test_concurrency.py

```
@@ -0,0 +1,246 @@
+
+    def test_concurrent_publish(self):
+        self.set_up()
+        config = ContextConfig()
+        config.units = self.units
+        context = self.create_test_context(config, self.data_future_ready)
+        self.assertTrue(context.is_ready())
+
+        publish_count = [0]
+        lock = threading.Lock()
+
+        def counting_publish(event):
```

Ruff (0.15.2)

[warning] Unused function argument: `event`

(ARG001)